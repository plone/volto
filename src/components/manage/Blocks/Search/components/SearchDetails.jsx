import React from 'react';
import { FormattedMessage, defineMessages, useIntl } from 'react-intl';

const messages = defineMessages({
  searchResults: {
    id: 'Search results',
    defaultMessage: 'Search results',
  },
});

const SearchDetails = ({ total, text, as = 'h4', data }) => {
  const El = as;
  const intl = useIntl();
  return (
    <El className="search-details">
      <>
        {text && (
          <>
            <FormattedMessage
              id="Searched for: <em>{searchedtext}</em>."
              defaultMessage="Searched for: <em>{searchedtext}</em>."
              values={{
                em: (...chunks) => <em>{chunks}</em>,
                searchedtext: text,
              }}
            />
          </>
        )}
        {data.showTotalResults && (
          <>
            {' '}
            {intl.formatMessage(messages.searchResults)}: {total}
          </>
        )}
      </>
    </El>
  );
};

export default SearchDetails;
