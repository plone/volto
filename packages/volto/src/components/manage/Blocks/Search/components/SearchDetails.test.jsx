import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import SearchDetails from './SearchDetails';

const mockStore = configureStore();

describe('SearchDetails', () => {
  it('renders search detail component', () => {
    const store = mockStore({
      userSession: { token: null },
      content: { data: { '@id': 'myid' } },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <SearchDetails
          text="test"
          total="10"
          data={{ showTotalResults: true }}
        ></SearchDetails>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
