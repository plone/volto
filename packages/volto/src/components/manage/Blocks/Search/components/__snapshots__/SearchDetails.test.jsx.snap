// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchDetails > renders search detail component 1`] = `
<h4
  className="search-details"
>
  Searched for: 
  <em>
    test
  </em>
  .
   
  Search results
  : 
  10
</h4>
`;
